src/libsystemd/sd-device/sd-device.c

```diff
@@ -1872,6 +1872,10 @@ _public_ int sd_device_get_sysattr_value(sd_device *device, const char *sysattr,
         assert_return(device, -EINVAL);
         assert_return(sysattr, -EINVAL);
 
+        /* Refuse to read uevent file by using this function. */
+        if (streq(sysattr, "uevent"))
+                return -EINVAL;
```

Comment:
I don't think we want to break compatibility like that. Even if no publicly visible code does that, sd-device has been around for a while and we don't know all the users.

Reply:
Indeed. Only we need to do is disabling caching when writing uevent files. I have no idea why I decided to add this yesterday :-)